.github/workflows/dotnet.yml

```diff
@@ -0,0 +1,33 @@
+# This workflow will build a .NET project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-net
+
+name: .NET
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 7.0.x
+    - name: Restore dependencies
+      run: dotnet restore
+    - name: Build
```

Comment:
Why not like [this](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-net#using-the-net-starter-workflow)
```
- name: Build
   run: dotnet build --configuration Release --no-restore
 - name: Test
    run: dotnet test --no-restore --verbosity normal
```
?

Reply:
I will add --configuration Release. 
I don't know why should we change -no-build to -no-restore in thest since we don't need to build again.

Reply:
with no-build it failed so i changed it to no-restore like you suggestd